src/node_crypto.cc

```diff
@@ -4249,32 +4137,32 @@ void DiffieHellman::ComputeSecret(const FunctionCallbackInfo<Value>& args) {
   }
 
   ClearErrorOnReturn clear_error_on_return;
-  BIGNUM* key = nullptr;
 
   if (args.Length() == 0) {
     return THROW_ERR_MISSING_ARGS(
         env, "Other party's public key argument is mandatory");
-  } else {
-    THROW_AND_RETURN_IF_NOT_BUFFER(env, args[0], "Other party's public key");
-    key = BN_bin2bn(
-        reinterpret_cast<unsigned char*>(Buffer::Data(args[0])),
-        Buffer::Length(args[0]),
-        0);
   }
 
-  int dataSize = DH_size(diffieHellman->dh);
+  THROW_AND_RETURN_IF_NOT_BUFFER(env, args[0], "Other party's public key");
+  BignumPointer key(BN_bin2bn(
+      reinterpret_cast<unsigned char*>(Buffer::Data(args[0])),
+      Buffer::Length(args[0]),
+      0));
+
+  int dataSize = DH_size(diffieHellman->dh_.get());
   char* data = Malloc(dataSize);
```

Comment:
Maybe use automatic memory management for this as well (and rename `dataSize` to `data_size` while you're here)?  Just a suggestion.

Reply:
Good idea – I’ve switched this to `MallocedBuffer`, too (that takes care of `dataSize` anyway)